### ISSUE TITLE
在 SecretFlow 拆分学习中实现 CAFE 防御

### ISSUE THREAD

**Issue by Yeekin-GYJ** [secretflow/secretflow#1241]
>此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP)](https://github.com/orgs/secretflow/discussions/1181)任务 ISSUE，欢迎社区开发者参与共建～
若有感兴趣想要认领的任务，但还未报名，辛苦先完成[报名](https://www.wjx.top/vm/QKio0dq.aspx#)进行哈～
  
## 任务介绍
- **任务名称**：在 SecretFlow 拆分学习中实现 CAFE 防御
- **技术方向**：FL/security
- **任务难度**：进阶🌟🌟

## 详细要求
- 功能性：在 SecretFlow 中实现论文[Cafe: Catastrophic data leakage in vertical federated learning](https://proceedings.neurips.cc/paper_files/paper/2021/file/08040837089cdf46631a10aca5258e16-Paper.pdf)的 CAFE 防御算法。需要使用 Callback 框架实现。
- 代码规范：Python 代码需要使用 black+isort 进行格式化（流水线包含代码规范检查卡点）
- 框架后端：优先实现 PyTorch 版本，同时实现 TensorFlow 版本更好。
- 提交说明：关联该 isuue 并提交代码至 https://github.com/secretflow/secretflow/blob/main/secretflow/ml/nn/sl/defenses

## 能力要求
- 了解基本 git 操作
- 熟悉 深度学习、联邦学习
- 对 SecretFlow 有一定了解

## 操作说明
- 范例：[PyTorch 后端的 gradient average 防御](https://github.com/secretflow/secretflow/blob/main/secretflow/ml/nn/sl/defenses/gradient_average.py)

**Comment by Chi-Kai** [comment]
Chi-Kai Give it to me

**Comment by Yeekin-GYJ** [comment]
> 奇凯 给我

感谢认领呀，任务完成过程中有遇到问题可以在 issue 上直接反馈，也可以添加小助手微信：SecretFlow04 进行沟通 加入 OSCP 专属群聊，期待您的PR～

**Comment by Yeekin-GYJ** [comment]
该任务与 [secretflow #1238](https://github.com/secretflow/secretflow/issues/1238)、[secretflow #1239](https://github.com/secretflow/secretflow/issues/1239)、[secretflow #1240](https://github.com/secretflow/secretflow/issues/1240) 为「开源之夏 2024」[SecretFlow 拆分学习中实现两个攻击及两个防御算法](https://summer-ospp.ac.cn/org/prodetail/24e430515?list=org&navpage=org)的项目课题，不可单独认领。具体认领方式请遵循「开源之夏 2024」活动规则～详情查看：https://summer-ospp.ac.cn/help/student/

**Comment by Rainysponge** [comment]
Rainysponge Give it to me
